lib/parallel_tests/grouper.rb

```diff
@@ -46,6 +51,47 @@ def in_even_groups_by_size(items, num_groups, options = {})
 
       private
 
+      def specify_groups(items, num_groups, options, groups)
+        specify_test_process_groups = options[:specify_groups].split('|')
+        if specify_test_process_groups.count > num_groups
+          raise 'Number of processes separated by pipe must be less than or equal to the total number of processes'
+        end
+
+        all_specified_tests = specify_test_process_groups.map { |group| group.split(',') }.flatten
+        specified_items_found, items = items.partition { |item, _size| all_specified_tests.include?(item) }
+
+        specified_specs_not_found = all_specified_tests - specified_items_found.map(&:first)
+        if specified_specs_not_found.any?
+          raise "Could not find #{specified_specs_not_found} from --specify-groups in the main selected files & folders"
+        end
+
+        if specify_test_process_groups.count == num_groups && items.flatten.any?
+          raise(
+            "The number of groups in --specify-groups matches the number of groups from -n but there were other specs " \
+            "found in the main selected files & folders not specified in --specify-groups. Make sure -n is larger than the " \
+            "number of processes in --specify-groups if there are other specs that need to be run. The specs that aren't run: " \
+            "#{items.map(&:first)}"
+          )
+        end
+
+        # First order the specify_groups into the main groups array
+        specify_test_process_groups.each_with_index do |specify_test_process, i|
+          groups[i] = specify_test_process.split(',')
+        end
+
+        # Return early here as we've processed the specify_groups tests and those exactly match the items passed in
```

Comment:
```suggestion
        # Return early when processed specify_groups tests exactly match the items passed in
```

Reply:
... should there be an error when there are tests left over though ?
as in: I specify 4 groups but there is 1 test left over that I did not specify -> would not run ... so it should fail ?

Reply:
if you pass in -n 4, and there are 4 specify groups, and there are extra tests, you will be returned in the earlier return on line 69

Here's ive described the different scenarios

### specify group # of processes equal n
* extra items from main selected files/folders
    * return early because some test will not be run
* no extra items
    * simply run all specify group tests. This means all tests in specify group exactly match items (this is the early return here)

### specify group # of processes are less than n
* extra items from main selected files/folders
    * fine
* no extra items
    * should also be fine. It’s the same thing as if you call bundle exec parallel_rspec -n 2 spec/app_spec.rb. it will still run the 1 test

Reply:
thx for the details, that sounds right! :)